### ISSUE TITLE
pip fails when installing the [test] dependencies

### ISSUE THREAD

**Issue by sergisiso** [stfc/PSyclone#881]
On a new installation in Ubuntu 20.04, when doing a:
``` pip install -e .[test]```
I get the error `ERROR: pytest-pylint 0.17.0 has requirement pylint>=2.3.0, but you'll have pylint 1.6.5 which is incompatible.`

This is because `setup.py` requests a specific version of pylint `"pylint==1.6.5"`, if I remove the specific version or the whole pylint dependency and let it get it as part of pytest-pylint the problem is solved and everything seems to work fine.

I couldn't find any issue regarding why pylint is locked to a specific version. Does anybody know? Is it safe to unlock it now?

**Comment by rupertford** [comment]
I think this was due to an error in one of the versions of pylint that led to recursion errors. However, we're way beyond that version now so I'm sure it's OK to remove the specific version.

I also had problems with Ubuntu 20.04 but in my case it was the move to python3 (I guess because I am not using a virtual env). I could get away with an alias for python=python3 in most cases but for the examples I had to create a symbolic link from the non-existent python distribution to the python3 distribution. Even now I am having to use pytest-3 explicitly.
